src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -288,30 +288,133 @@ void FilterInvalidFunctions(policy_table::Rpc& rpcs) {
 }
 
 /**
- * @brief FilterInvalidParameters filter parameters that not present in schema
+ * @brief FilterInvalidRPCParameters filter parameters that not present in
+ * schema
  * @param rpc_parameters parameters to filter
  */
-void FilterInvalidParameters(policy_table::RpcParameters& rpc_parameters) {
+void FilterInvalidRPCParameters(policy_table::RpcParameters& rpc_parameters) {
+  policy_table::HmiLevels valid_hmi_levels;
+  for (const auto& hmi_level : rpc_parameters.hmi_levels) {
+    if (hmi_level.is_valid()) {
+      valid_hmi_levels.push_back(hmi_level);
+    }
+  }
+  rpc_parameters.hmi_levels.swap(valid_hmi_levels);
+
   policy_table::Parameters valid_params;
-  for (auto& param : *(rpc_parameters.parameters)) {
+  for (const auto& param : *(rpc_parameters.parameters)) {
```

Comment:
@JackLivio I would suggest you to init const reference to `*(rpc_parameters.parameters)` and use it in the loop